### ISSUE TITLE
expend functionality to match independent and internal IDs when corresponding

### ISSUE THREAD

**Issue by thea-m** [BetaMasaheft/Documentation#811]
From discussion with @eu-genia :
There is some confusion in our records as to when works that circulate independently and are also part of larger works should be referred to by their independent or derived ID.
For example, the Anaphoras contained in LIT1960Mashaf all have their independent ID and a derived ID in LIT1960Mashaf (for example LIT1099Anapho = LIT1960Mashaf#AnaphoraMary). Now in manuscripts containing LIT1960Mashaf, both LIT1960Mashaf#AnaphoraMary and LIT1099Anapho have be used to refer to the same Anaphora in a similar context, as part of LIT1960Mashaf (for example LIT1960Mashaf#AnaphoraMary in FSUor15 and LIT1099Anapho in BAVet15). 
Which would be the preferable way of encoding here? Should the independent ID always be added in @corresp so that both are given?
If both ways are fine, it would be very useful if all information related to one of the IDs could be retrieved in the other, if this is possible at all  (for example, if all manuscripts containing LIT1960Mashaf#AnaphoraMary would also be displayed in the comparison view of LIT1099Anapho; or even if in PRS3251Cyriacus, a reference would be made automatically to LIT1960Mashaf#AnaphoraMary, since there exists a relation between this person and LIT1099Anapho).

**Comment by PietroLiuzzo** [comment]
Systematization of this referencing can be achieved now by associating systematically both forms of identification, even if an explicit relation is not present

**Comment by thea-m** [comment]
I'm not sure whether I understood your last comment (or whether it was meant only for yourself, in which case, wonderful). 
Regarding the two manuscripts, they differ, but this is to be seen in relation to normal variance of Missals (of which our understanding is not great and the encoding not exemplary). I do not read such a statement (anaphora added independently in BAVet15) in the record.

**Comment by PietroLiuzzo** [comment]
I wish I had been less elliptic... I remember distinctly doing some work on this. I do not find that work again anyhow and I will do it again

**Comment by PietroLiuzzo** [comment]
I think the possibility of using one or the other is one I would not give up in favour of a rule for one or the other identifier, although priority to the independent ID is of course easier in most cases, so I would stay with my previous statement that using one or the other form of identification is actually a meaningfull distinction which should stay. 
I would use the LIT1960Mashaf#AnaphoraMary for for example if a fragment is at hand which is clearly of a manuscript which contained the entire LIT1960Mashaf but where the extant part includes LIT1960Mashaf#AnaphoraMary complete or incomplete, so that my msContent can have the references to the exact portions left. 
if in a complete manuscript LIT1099Anapho is present outside of  LIT1960Mashaf I would prefer that id. 
In a manuscript of LIT1960Mashaf where this is complete the only thing I really need to add is this id, if I want to add the subparts because of an altered order, then I would go for LIT1960Mashaf#AnaphoraMary. Bottom line: I think the cataloguer should be free to choose and the two things are not identical, and the corresponding statement says just what it should say nicely, but this is really not for me to regulate, I can only give you my tuppence... this procedural description and decision making support should be 
- [ ] documented in the guidelines

Searching for already supported functionality in matching e.g. LIT1099Anapho and LIT1960Mashaf#AnaphoraMary, which I did not document enough I did the following
- [x] the list of manuscripts for LIT1099Anapho does already include an additional listing of manuscripts in which works which contain it, i.e. have a `t:div[@type eq 'textpart'][@corresp eq  LIT1099Anapho]`. This IMHO is better than a mixed list of all possibilities. I have added a count of the manuscripts in the second list, also because it breaks down the eventuality that such unit which is also a part of another is contained in more than one textual unit, e.g. a LIT9999notyetexistingwork#maryAnaph exists which is a record  LIT9999notyetexistingwork containing `t:div[@type eq 'textpart'][@corresp eq  LIT1099Anapho][@xml:id eq  'maryAnaph']`. 
- [x] it is true that if I navigate to LIT1960Mashaf I remain unaware of witnesses of its parts which have a corresponding independent id. I have added a further sublist, listing the witnesses of any value of `t:div[@type eq 'textpart']/@corresp`, `t:relation[@name eq 'saws:contains']/@passive`  `t:relation[(@name eq 'saws:formsPartOf') or (@name eq 'ecrm:CLP46i_may_form_part_of')]/@active` preceded by the explanation that parts of the TU have are also independent TUs attested in mss as such. the list for LIT1960Mashaf goes up to 50 parts, for each of which a list of titles is given. I believe this is good navigational hint and exposes what the db knows correctly, without mixing what should not be mixed.
- the two features above in the main landing pages for TUs, allow the decision on what to use, and how to be taken without further impact on the view, which exposes now what was hidden and consequently confusing.
I would not extend results in other views, exactly for the reasons above, that is to say, to keep evident the distinction and relation, without silently mixing contents. 
- [x] the comparison functionality does not support more than one ID, I have diversified the inputs of the hidden compareSelected function, so that on top of comparing diverse manuscripts it supports also comparing witnesses of multiple Textual Units, that is to say, both LIT1960Mashaf#AnaphoraMary and LIT1099Anapho can be passed to it, and the db will select manuscript which contain one or the other and list all. I have added a link to this functionality in the same manuscript box.
